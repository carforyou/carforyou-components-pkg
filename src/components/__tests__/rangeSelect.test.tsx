import React from "react"
import { fireEvent, render } from "@testing-library/react"

import RangeSelect from "../rangeSelect"

const onChangeMin = jest.fn()
const onChangeMax = jest.fn()

const renderWrapper = ({
  name = "name",
  minValue = null,
  maxValue = null,
  minPlaceholder = "min",
  maxPlaceholder = "max",
  handleChangeMin = onChangeMin,
  handleChangeMax = onChangeMax,
  options = [
    { name: "One", value: 1 },
    { name: "Two", value: 2 },
    { name: "Three", value: 3 },
    { name: "Four", value: 4 },
    { name: "Five", value: 5 },
    { name: "Six", value: 6 },
  ],
} = {}) =>
  render(
    <RangeSelect
      name={name}
      value={{
        min: minValue,
        max: maxValue,
      }}
      placeholder={{
        min: minPlaceholder,
        max: maxPlaceholder,
      }}
      handleChange={{
        min: handleChangeMin,
        max: handleChangeMax,
      }}
      options={options}
    />
  )

describe("RangeSelect", () => {
  it("renders RangeSelect correctly", () => {
    const { container } = renderWrapper()
    expect(container).toMatchSnapshot()
  })

  it("renders options on click", () => {
    const { getByText } = renderWrapper()

    fireEvent.click(getByText("max"))
    expect(getByText("One"))
  })
})
