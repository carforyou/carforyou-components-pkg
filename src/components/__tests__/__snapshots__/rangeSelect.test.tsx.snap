// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSelect renders RangeSelect correctly 1`] = `
<div>
  <div>
    <div
      class="inline-block w-6/12"
    >
      <div
        class="w-12/12 relative"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby=""
          class="w-12/12"
          role="combobox"
        >
          <button
            aria-haspopup="true"
            aria-label="open menu"
            class="w-12/12 cursor-pointer transition duration-200 focus:outline-none"
            data-testid="dropdown-toggle"
            data-toggle="true"
            role="button"
            type="button"
          >
            <div
              class="relative pb-px"
            >
              <div
                class="validation-inputContainer mb-20"
              >
                <div
                  class="relative flex w-12/12 clearButton-container"
                >
                  <div
                    class="w-12/12 text-left select-toggle input_withClearButton input_left hover:z-1 hover:transition hover:duration-200 focus:z-1 select bg-white select_closed text-grey-4"
                    data-valid="true"
                  >
                    min
                  </div>
                </div>
              </div>
              <div />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="inline-block w-6/12"
    >
      <div
        class="w-12/12 relative"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby=""
          class="w-12/12"
          role="combobox"
        >
          <button
            aria-haspopup="true"
            aria-label="open menu"
            class="w-12/12 cursor-pointer transition duration-200 focus:outline-none"
            data-testid="dropdown-toggle"
            data-toggle="true"
            role="button"
            type="button"
          >
            <div
              class="relative pb-px"
            >
              <div
                class="validation-inputContainer mb-20"
              >
                <div
                  class="relative flex w-12/12 clearButton-container"
                >
                  <div
                    class="w-12/12 text-left select-toggle input_withClearButton input_right -ml-1 hover:z-1 hover:transition hover:duration-200 focus:z-1 select bg-white select_closed text-grey-4"
                    data-valid="true"
                  >
                    max
                  </div>
                </div>
              </div>
              <div />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
